audeer/core/io.py

```diff
@@ -86,6 +87,46 @@ def all_names_equal(name):
     ))
 
 
+def create_archive(
+        root: str,
+        files: typing.Union[str, typing.Sequence[str]],
+        archive: str,
+):
+    r"""Create ZIP or TAR.GZ archive.
+
+    Args:
+        root: path to root folder of archive.
+            Path names inside the archive
+            will be relative to ``root``
+        files: files to include in archive,
+            relative to ``root``
+        archive: path to archive file.
+            The archive type is determined by the file extension
+
+    Raises:
+        RuntimeError: if archive does not end with ``zip`` or ``tar.gz``
+
+    """
+    archive = safe_path(archive)
+    mkdir(os.path.dirname(archive))
+    files = to_list(files)
+    if archive.endswith('zip'):
+        with zipfile.ZipFile(archive, 'w', zipfile.ZIP_DEFLATED) as zf:
+            for file in files:
+                full_file = os.path.join(root, file)
+                zf.write(full_file, arcname=file)
+    elif archive.endswith('tar.gz'):
+        with tarfile.open(archive, "w:gz") as tf:
+            for file in files:
+                full_file = os.path.join(root, file)
```

Comment:
should we check if file exists?

Reply:
I think it should be fine if we leave this to `tarfile`